developer/styleguide.md

```diff
@@ -1,119 +1,216 @@
-    social/
-    taxes-avalara/
+### Naming variables
 
-    // All other folder names everywhere
-    addressBook/
+Variable names should be:
 
-**Bad**
+- Meaningful. Avoid single-letter variables.
+- Use `error`, instead of `e` or `err`
+- Variables that contains a boolean value should use a `isThisTrue` style.
+- Variables that return arrays should be plural.
 
-- Package name should contain hyphens to make it easier to read.
-- Underscores are not to be used unless expressly required.
+Publication names should:
 
+- Use TitleCase
 
-    reactionpackagename/
-    address_book/
+### Working with variables
 
-#### File Names
+- Use `const` except when a variable is being reassigned.
 
-**Good**
+## Methods
 
-- File names must start with a lowercased letter and may be camel cased if joining more than one word.
-- File names may contain multiple (.) characters as needed
+### Method names should:
+- Methods names should use a verb/noun style when possible, e.g. `checkConfiguration` or `addToCart`.
+- Methods that return a single value should be singular.
+- Methods whose main purpose is to return a value should use the get prefix.
 
+- Avoid ternary operators and one-line `if` statements
 
-    settingsContainer.js
-    publishContainer.js
-    addressBook.js
-    bootstrap.rtl.js
-    index.js
+### Use parenthesis for clarity
 
-    // This is an exception as it's part of Meteor's naming convention
-    addressBook.app-test.js
+- Use parenthesis around all method arguments.
 
-**Bad**
+**Don't**
+```
+cartItems.find(item => item.status === picked)
+```
+
+**Do**
+```
+cartItems.find((item) => item.status === picked)
+```
+
+- When specifying a callback, always use the parenthesis to indicate the argument being accepted over the bare arrow-function form. This way, it's clear it's not another argument to the original function:
+
+**Don't**
+```
+thisMethodTakesACallback(arg1, arg2, result => {
+  doStuff
+});
+```
+**Do**
+
+```
+thisMethodTakesACallback(arg1, arg2, (result) => {
+  dostuff();
+});
+```
+
+
+### Avoid multi-line shorthand arrow functions
+
+If you have to break your arrow function into multiple lines, use curly brackets and a return rather than trying to break the shorthand arrow function into multiple lines.
+
+
+**Don't**
+ ```
+cartItems.find(
+   (item) => item.status === status
+ )
+```
+
+**Do**
+ ```
+cartItems.find((item) => {
+  return item.status === status;
+})
+```
+
+### Working with collections
+
+Be explicit in querying:
+
+**Don't**
+```
+Products.findOne("abc123")
+```
+**Do**
+```
+Products.findOne({ _id: "abc123" })
+```
 
-- Hyphens and underscores are not to be used unless expressly required; such is the case with Meteor for `*.app-test.js` files. Or for migration files to make the filename more readable.
+### Replacing lodash with ES2015
 
+Use native ES6 elements over lodash whenever there is a 1-for-1 replacement. Here are some examples of lodash methods that can be replaced with native elements:
```

Comment:
Should probably decide if we're going to say ES6 or ES2015 and be consistent there.

Also many of these Array.prototype methods have been around since well before ES6

Reply:
I think we might just want to just say "native Javascript"?